proto/storage/node.proto

```diff
@@ -86,11 +87,47 @@ message NodeScan {
 }
 
 message NodeInventory {
-    string                         node_id    = 1 [(gogoproto.moretags) = 'search:"Node ID,store" sql:"pk,type(uuid)"'];
-    string                         node_name  = 2 [(gogoproto.moretags) = 'search:"Node,store"'];
-    google.protobuf.Timestamp      scan_time  = 3 [(gogoproto.moretags) = 'search:"Node Scan Time,store"'];
-    scannerV1.Components           components = 4;
-    repeated scannerV1.Note        notes      = 5;
+    string                         node_name  = 1;
+    google.protobuf.Timestamp      scan_time  = 2;
+
+    // Components represents a subset of the scannerV1.Components proto message containing only fields required for RHCOS node scanning
+    // We are not using scannerV1.Components here for the following reasons:
+    // - to avoid conflicts between v1 and scannerV1 APIs when generating the code in central/graphql/resolvers/generated.go
+    // - to not expose scanner v1 API over stackrox graphql API
+
+    message Components {
+        string                                     namespace     = 1;
+        message RHELComponent {
+            int64               id          = 1;
+            string              name        = 2;
+            string              namespace   = 3;
+            string              version     = 4;
+            string              arch        = 5;
+            string              module      = 6;
+            repeated string     cpes        = 7;
+            string              added_by    = 8;
+            message Executable {
+                string                      path              = 1;
+                message FeatureNameVersion {
+                    string name    = 1;
+                    string version = 2;
+                }
+                repeated FeatureNameVersion required_features = 2;
+            }
+            repeated Executable executables = 9;
+        }
```

Comment:
As part of [ROX-13794](https://issues.redhat.com/browse/ROX-13794), we will not map CPEs in the node "invetorization" (aka. scanning). Instead, we will return the content sets found in the O.S. installation. The CPE mapping will be performed by Central Scanner when matching vulnerabilities. Because of that, we need to replace `cpe` with `content_sets`. Two options:

1. Replace the `cpe` field in `RHELComponent` with `content_sets`.
2. Remove the `cpe` field and add a `content_sets` field to `Component`.

Option 2 exists because today, we cannot determine precisely which content set each package came from, so we assume the package might have come from any of the content sets specified in the O.S. installation. This may be possible in the long future, so we might just use individual fields just as we do with CPEs, especially considering the size is neglectable.

CC @vikin91

Reply:
See related PR: https://github.com/stackrox/scanner/pull/1053

Reply:
Thanks! As #4392 is a prefactor to this PR, the change will be addressed in #4392 (option 2) and then this PR will be rebased.